src/Adapter/Shop/MaintenanceConfiguration.php

```diff
@@ -27,21 +27,28 @@
 namespace PrestaShop\PrestaShop\Adapter\Shop;
 
 use PrestaShop\PrestaShop\Adapter\Configuration;
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Configuration\MultiStoreConfiguratorInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 
 /**
  * This class loads and saves data configuration for the Maintenance page.
  */
-class MaintenanceConfiguration implements DataConfigurationInterface
+class MaintenanceConfiguration implements MultiStoreConfiguratorInterface
 {
     /**
      * @var Configuration
      */
     private $configuration;
 
-    public function __construct(Configuration $configuration)
```

Comment:
Doesn't this count as BC break? (Not saying that it shouldn't be done just that it should be noted in PR description.).